Add roles

markcmhill opened the pull request:
Following on from previous pull request, all log ins now handled through user class with roles (artist, subscriber) defined in a separate model.

Commit:
moved the artist signup into the signup controller

Commit:
place artist and user signup on the same page

Commit:
removed artist sign-up from nav bar

Commit:
combine user and artist front end sign up controllers

Commit:
work in progress refactoring back end controllers

Commit:
front end changes work in progress

Commit:
combined user and artist registration in back end

Commit:
user and artist can register but not all tests passing

Commit:
created a local storage service on the front end for storing account information returned from the back end

Commit:
log in now uses local storage service

Commit:
deleted unnecessary controller

Commit:
artist and user log in handled through the same controller

Commit:
passing tests for artist sign in up

Commit:
changed nav bar to only display option to upload tracks when artist is logged in

Commit:
WIP

Commit:
feature tests passing for log in and out

yanyil committed:
fix bugs in streams, tracks models

yanyil committed:
delete tracks folder

Commit:
feature tests for calculating artist payments passing, added id to song images to facilitate capybara testing

Commit:
in feature tests for playing tracks changed instructions to click button to click image by id

Commit:
added role model which belongs to user

Commit:
added methods to check for roles

Commit:
can sign up using roles

Commit:
login with roles working

Commit:
removed console log

Commit:
added name to users and allow balance to be displayed

Commit:
restored styles

yanyil commented on back-end/app/controllers/registrations_controller.rb:
```diff
@@ -1,12 +1,21 @@
+
+  private
+  def new_account
+      return User.new(name: params[:name], email: params[:email], password: params[:password], password_confirmation: params[:password_confirmation])
```
no need for return as implicit?

Commit:
Delete Project_-_23_03_2016__13.47.mp3

Commit:
added unit tests for backend controllers and models

Commit:
user and listener sign up and log in are working with feature tests passing

Commit:
playing tracks, recording streams, and payments from listener to artist working

Commit:
removed spec folder from coveralls

Commit:
Added validation for Listeners and Artist

- Listeners cannot login from the Artist login tab and vice-versa
- Added unit tests for the sessions controllers
- Created sessions_helper while refactoring unit test

Commit:
Renamed templates files and added validations

- Renamed htmls files to avoid confusion
- Restored 2 pictures in the carousel controller
- Added ng-click to submit buttons in both artist and listener login tabs

Commit:
Added public system directory to git ignore to prevent files being added to git

Commit:
Merge branch 'add-roles' of https://github.com/MarcoCode/fairMuse into contributions

Commit:
Added ng-click values to sign in buttons in order to validate log in; changed names in tests to reflect new css class names; ensured that artist parameter was being sent by log in controller

MarcoCode commented:
:shipit:
